### DISCUSSION TITLE
make default log timeformat human readable

### DISCUSSION THREAD

**Discussion by ArgTang** [rabbitmq/messaging-topology-operator#259] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
`{"level":"info","ts":1632735489.3334155, ...}`
the ts field in the logs is not human readable. 

**Describe the solution you'd like**
The logs print timestamp with ex. iso8601 formatting

**Comment by ansd** [comment]
I think the default time format of the JSON logs should stay a Unix time stamp because
1. That's what's defined in the [Kubernetes JSON log format](https://kubernetes.io/docs/concepts/cluster-administration/system-logs/#json-log-format): `ts - timestamp as Unix time (required, float)`
2. That's the default of the [zap production mode logger used in controller runtime](https://github.com/kubernetes-sigs/controller-runtime/tree/master/pkg/log/zap).
3. Above 2 points seem to me a very good default because in production, you'll have a logging backend. Every decent logging backend (or even earlier on the log shipper) can convert the time format from unix timestamp to whatever human readable time format you prefer.
4. If you don't run the operator in production, use the `--zap-devel` command line flag in the operator deployment. It will print more human-readable output, i.e. no JSON and human readable time format, e.g. `2021-09-30T10:08:41.229Z	INFO	controller-runtime.manager.controller.rabbitmqcluster	Finished reconciling ...`

**Comment by ArgTang** [comment]
Since there is no helm chart available and its no configuration section in the readme it is not easy to discover these flags. Can you change the loglevel as well here?

yes we do have some default logging pipeline setup. We have some parsing rules with fluent bit, but we prefer human readable timestamps. We can usually configure it and have not had any apps force epoch before. Another point is when looking directly at the logs in the pod as well make it hard to parse the timestamp. to see what logs is relevant.

**Comment by ansd** [comment, marked as answer]
>Can you change the loglevel as well here?

The command line flags of the zap logger can be found [here](https://github.com/kubernetes-sigs/controller-runtime/blob/66537ca5b7439b06f2f3b08901640f934834c9a1/pkg/log/zap/zap.go#L227-L233).

For example, to only output error logs, configure the messaging-topology-operator deployment YAML as follows:
```yaml
    spec:
      containers:
      - args:
        - --zap-log-level=error
        command:
        - /manager
```

I agree, we should document these flags.

>Another point is when looking directly at the logs in the pod as well make it hard to parse the timestamp. to see what logs is relevant.

You can use `kubectl logs --timestamps`. This flag will include human readable timestamps on each line in the log output (They might deviate by a few microseconds from the timestamp output by the RabbitMQ logs / Unix time. However, that should be good enough in most cases.)

**Comment by ArgTang** [comment]
Thanks for the tips. we'll see if we can work around this

**Comment by Zerpet** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(
